### PR TITLE
Optimize Snowflake load_file using native COPY INTO

### PR DESCRIPTION
**Pull request by tatiana**
Fix: #430 

Reduced the time to load to Snowflake by 20% for 5GB datasets (from 24.46 min to 5.49 min). Further details in the PR results file.

Co-authored-by: Ankit Chaurasia <ankit.chaurasia@astronomer.io>

### CODE REVIEW

File: src/astro/databases/snowflake.py [hunk 1]
```diff
@@ -291,6 +292,52 @@ def drop_stage(self, stage: SnowflakeStage) -> None:
...
+        - The user must have permissions to create a STAGE in Snowflake.
+        - If loading from GCP Cloud Storage, `native_support_kwargs` must define `storage_integration`
+        - If loading from AWS S3, the credentials for creating the stage may be
+        retrieved from the Airflow connection or from the `storage_integration`
+        attribute within `native_support_kwargs`.
+
+        :param source_file: File from which we need to transfer data
+        :param target_table: Table to which the content of the file will be loaded to
+        :param if_exists: Strategy used to load (currently supported: "append" or "replace")
+        :param native_support_kwargs: kwargs to be used by method involved in native support flow
```
Comment (sunank200) [inline src/astro/databases/snowflake.py:318]: Would be it useful for users to know what are different `native_support_kwargs` supported? or may be reference to `COPY INTO` documentation?

**Review by sunank200 — APPROVED** [review]
Looks good to me with a minor comment.

File: tests/databases/test_snowflake.py [hunk 2]
```diff
@@ -450,3 +450,40 @@ def test_create_stage_amazon_fails_due_to_no_credentials(get_credentials):
...
+        {"provider": "amazon", "filetype": FileType.CSV},
+    ],
+    indirect=True,
+    ids=["amazon_csv"],
+)
+def test_load_file_to_table_natively(remote_files_fixture, database_table_fixture):
+    """Load a file to a Snowflake table using the native optimisation."""
+    filepath = remote_files_fixture[0]
+    database, target_table = database_table_fixture
+    database.load_file_to_table(
```
Comment (sunank200) [inline tests/databases/test_snowflake.py:476]: This test should have called `load_file_to_table_natively()`.

**Review by sunank200 — CHANGES_REQUESTED** [review]
The test is calling wrong method. Thats the reason coverage shows this test not being run.

File: tests/databases/test_snowflake.py (hunk 2 above)
Comment (sunank200) [reply tests/databases/test_snowflake.py:476]: Makes sense

**Review by sunank200 — APPROVED** [review]

File: src/astro/databases/snowflake.py (hunk 1 above)
Comment (tatiana) [reply src/astro/databases/snowflake.py]: @sunank200, please, have a look now!
I had already added the reference to the official docs on the bottom of the docstring, do you think I should add some other? Or place it in another place?

**Review by kaxil — APPROVED** [review]

**Comment by tatiana** [comment]
We added the benchmark results to the PR and fixed the broken tests.

Two checks are failing probably because of https://github.com/pypa/pip/pull/11298 - which affects all branches. I just re-run the tests for a branch which was passing 4h ago, that wasn't changed and the same tests failed.

I've contacted TP to get his advice, and in the meantime, we'll merge this PR.

File: example_dags/example_amazon_s3_snowflake_transform.py [hunk 3]
```diff
@@ -21,17 +21,17 @@ def combine_data(center_1: Table, center_2: Table):
 @aql.transform()
 def clean_data(input_table: Table):
     return """SELECT *
-    FROM {{input_table}} WHERE TYPE NOT LIKE 'Guinea Pig'
+    FROM {{input_table}} WHERE type NOT LIKE 'Guinea Pig'
     """
 
 
-@aql.dataframe(identifiers_as_lower=False)
+@aql.dataframe()
```
Comment (tatiana) [inline example_dags/example_amazon_s3_snowflake_transform.py:28]: We should be able to handle capitalization more consistently across dataframes / loading file after implementing:
https://github.com/astronomer/astro-sdk/issues/564

File: example_dags/example_amazon_s3_snowflake_transform.py [hunk 4]
```diff
@@ -67,11 +67,11 @@ def example_amazon_s3_snowflake_transform():
     )
 
     temp_table_1 = aql.load_file(
-        input_file=File(path=f"{s3_bucket}/ADOPTION_CENTER_1.csv"),
+        input_file=File(path=f"{s3_bucket}/ADOPTION_CENTER_1_unquoted.csv"),
```
Comment (tatiana) [inline example_dags/example_amazon_s3_snowflake_transform.py:70]: The CSV was very odd beforehand in the format:
```
"header1","header2"
"value1","value2"
```

The COPY INTO command wasn't acceptable in this format, and we decided that this file could be cleaned beforehand. Users may face similar issues in future.

File: tests/benchmark/results.md [hunk 5]
```diff
@@ -95,6 +94,25 @@ The benchmark was run as a Kubernetes job in GKE:
...
+* Container resource limit:
+  * Memory: 10 Gi
+
+| database   | dataset    | total_time   | memory_rss   | cpu_time_user   | cpu_time_system   |
+|:-----------|:-----------|:-------------|:-------------|:----------------|:------------------|
+| snowflake  | ten_kb     | 9.1s         | 56.45MB      | 2.56s           | 110.0ms           |
+| snowflake  | hundred_kb | 9.19s        | 45.4MB       | 2.55s           | 120.0ms           |
+| snowflake  | ten_mb     | 10.9s        | 47.51MB      | 2.58s           | 160.0ms           |
+| snowflake  | one_gb     | 1.07min      | 47.94MB      | 8.7s            | 5.67s             |
+| snowflake  | five_gb    | 5.49min      | 53.69MB      | 18.76s          | 1.6s              |
```
Comment (tatiana) [inline tests/benchmark/results.md:114]: I'll log a ticket for us to further investigate why we weren't able to run the benchmark with 10 GB for this implementation.

File: src/astro/databases/snowflake.py [hunk 6]
```diff
@@ -291,6 +301,93 @@ def drop_stage(self, stage: SnowflakeStage) -> None:
...
+        """
+        if file:
+            dataframe = file.export_to_dataframe(
+                nrows=settings.LOAD_TABLE_AUTODETECT_ROWS_COUNT
+            )
+
+        # Snowflake doesn't handle well mixed capitalisation of column name chars
+        # we are handling this more gracefully in a separate PR
+        if dataframe is not None:
+            dataframe.columns.str.upper()
```
Comment (tatiana) [inline src/astro/databases/snowflake.py:325]: We should be able to handle capitalization more consistently across dataframes / loading file after implementing: #564

**Pull request merged**